prov/sockets/src/sock_progress.c

```diff
@@ -2150,7 +2150,7 @@ static int sock_pe_progress_rx_pe_entry(struct sock_pe *pe,
 
 	if (sock_comm_is_disconnected(pe_entry)) {
 		SOCK_LOG_DBG("conn disconnected: removing fd from pollset\n");
-		if (pe_entry->conn->sock_fd != -1) {
+		if (pe_entry->ep_attr->cmap.used > 0) {
```

Comment:
I haven't looked at the code in detail.  That said, I wonder if it we can replace 'used' with something that carries more information, such as replacing it with a pointer or the fd, so that we can test whether the cmap is being used by the entry that we are wanting to remove.